src/ansys/pytwin/evaluate/tbrom.py

```diff
@@ -0,0 +1,270 @@
+        self._hasinfmcs = None
+        self._hasoutmcs = False
+
+        files = os.listdir(tbrom_path)
+        infdata = dict()
+        for file in files:
+            if TbRom.IN_F_KEY in file:
+                folder = file.split("_")
+                fname = folder[1]
+                inpath = os.path.join(tbrom_path, file, TbRom.TBROM_BASIS)
+                inbasis = TbRom._read_basis(inpath)
+                infdata.update({fname: inbasis})
+        self._infbasis = infdata
+
+        outpath = os.path.join(tbrom_path, TbRom.OUT_F_KEY, TbRom.TBROM_BASIS)
+        self._outbasis = TbRom._read_basis(outpath)
+        settingspath = os.path.join(tbrom_path, TbRom.OUT_F_KEY, TbRom.TBROM_SET)
+        [nsidslist, dimensionality, outputname, unit] = TbRom._read_settings(settingspath)
+        self._nsidslist = nsidslist
+        self._outdim = int(dimensionality[0])
+        self._outname = outputname
+        self._outunit = unit
+        self._outputfilespath = None
+
+    def points_generation(self, on_disk: bool, output_file_path: str, namedselection: str = None):
+        """
+        Generate a point file either in memory or on disk, for the full field or a specific part. If on disk, the point
+        file is written in the output_file_name path (currently no associated settings file is generated)
+
+        Parameters
+        ----------
+        on_disk: bool
+            Whether the point file is saved on disk (True) or returned in memory (False)
+        output_file_name: str
+            Path where the point file is written if on_disk is True
+        named_selection: str (optional)
+            Named selection on which the point file has to be generated
+        """
+        pointpath = os.path.join(self._tbrom_path, TbRom.OUT_F_KEY, TbRom.TBROM_POINTS)
+        vec = np.array(TbRom._read_binary(pointpath))
+        if namedselection is not None:
+            pointsids = self.namedselectionids(namedselection)
+            listids = []
+            for i in pointsids:
+                for k in range(0, 3):
+                    listids.append(i * 3 + k)
+            vec = vec[listids]
+        if on_disk:
+            TbRom._write_binary(output_file_path, vec)
+            return output_file_path
+        else:
+            return vec
+
+    def snapshot_generation(self, on_disk: bool, output_file_path: str, namedselection: str = None):
+        """
+        Generate a field snapshot based on current states of the TBROM, either in memory or on disk, for the full field
+        or a specific part. If on disk, the snapshot is written in the output_file_name path
+
+        Parameters
+        ----------
+        on_disk: bool
+            Whether the snapshot file is saved on disk (True) or returned in memory (False)
+        output_file_name: str
+            Path where the snapshot file is written if on_disk is True
+        named_selection: str (optional)
+            Named selection on which the snasphot has to be generated
```

Comment:
```suggestion
        on_disk : bool
            Whether the snapshot file is saved on disk (True) or returned in memory (False).
        output_file_name : str
            Path where the snapshot file is written when saved to disk.
        named_selection : str, optional
            Named selection to generate the snasphot on. The default is ``None``.
```
Earlier comments for these same parameters apply!